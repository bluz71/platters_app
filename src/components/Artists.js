// @flow
import React, { Component } from 'react';
import axios from 'axios';

const ARTISTS_URL = 'http://localhost:3000/artists.json';

class Artists extends Component {
  constructor(props) {
    super(props);

    this.state = {
      artists: []
    }
  }

  componentDidMount() {
    axios.get(ARTISTS_URL).then(response => {
      console.log(response);
    });
  }

  render() {
    return (
      <h2>Artists Page</h2>
    );
  }
}

export default Artists;
