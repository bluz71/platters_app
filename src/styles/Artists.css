.Artist {
  padding: 0.5em 0 0.5em 0.5em;
}

.Artist h2 {
  margin-top: 0.25em;
}

.Artist:hover {
  background: rgba(0, 0, 0, .025);
  border-radius: 0.25em;
}

.Artist a {
  text-decoration: none;
}

.Artists h1 {
  font-size: 1.58em;
}

.icon {
  font-size: 0.75em;
}
