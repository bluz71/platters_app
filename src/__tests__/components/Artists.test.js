import React from 'react';
import { shallow } from 'enzyme';
import Artists from '../../components/Artists';

// Details:
//   https://blog.rescale.com/testing-promise-side-effects-with-asyncawait
//   https://github.com/facebook/jest/issues/2157#issuecomment-279171856
const flushPromises = () => new Promise(resolve => setImmediate(resolve));

describe('<Artists />', () => {
  it('renders without crashing', () => {
    shallow(<Artists />);
  });

  it('renders div.Artists', () => {
    const wrapper = shallow(<Artists />);
    expect(wrapper.find('div.Artists').length).toEqual(1);
  });

  it('renders first page', async () => {
    const wrapper = shallow(<Artists />);
    await flushPromises();
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders second page', async () => {
    const wrapper = shallow(<Artists history={[]} />);
    // Manually trigger the callback
    wrapper.instance().handlePageChange(2);
    await flushPromises();
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders A artists', async () => {
    const wrapper = shallow(<Artists history={[]} />);
    // Manually trigger the callback
    wrapper.instance().handleLetter('A');
    await flushPromises();
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders ABC search match', async () => {
    const wrapper = shallow(<Artists history={[]} />);
    // Manually trigger the callback
    wrapper.instance().handleSearchChange('ABC');
    await flushPromises();
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });
});

