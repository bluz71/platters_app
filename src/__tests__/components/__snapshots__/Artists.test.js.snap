// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Artists /> renders A artists 1`] = `
<Row
  bsClass="row"
  componentClass="div"
>
  <Col
    bsClass="col"
    componentClass="div"
    md={10}
  >
    <div
      className="Artists"
    >
      <PageHeader
        bsClass="page-header"
      >
        Artists 
        <small>
          (
          1
           
          Artist
          )
        </small>
      </PageHeader>
      <div>
        <div
          className="filters"
        >
          <ul
            className="pagination pagination-sm"
          >
            <li
              onClick={[Function]}
            >
              <a>
                All
              </a>
            </li>
            <li
              className="active"
              key="0"
              onClick={[Function]}
            >
              <a>
                A
              </a>
            </li>
            <li
              key="1"
              onClick={[Function]}
            >
              <a>
                B
              </a>
            </li>
            <li
              key="2"
              onClick={[Function]}
            >
              <a>
                C
              </a>
            </li>
            <li
              key="3"
              onClick={[Function]}
            >
              <a>
                D
              </a>
            </li>
            <li
              key="4"
              onClick={[Function]}
            >
              <a>
                E
              </a>
            </li>
            <li
              key="5"
              onClick={[Function]}
            >
              <a>
                F
              </a>
            </li>
            <li
              key="6"
              onClick={[Function]}
            >
              <a>
                G
              </a>
            </li>
            <li
              key="7"
              onClick={[Function]}
            >
              <a>
                H
              </a>
            </li>
            <li
              key="8"
              onClick={[Function]}
            >
              <a>
                I
              </a>
            </li>
            <li
              key="9"
              onClick={[Function]}
            >
              <a>
                J
              </a>
            </li>
            <li
              key="10"
              onClick={[Function]}
            >
              <a>
                K
              </a>
            </li>
            <li
              key="11"
              onClick={[Function]}
            >
              <a>
                L
              </a>
            </li>
            <li
              key="12"
              onClick={[Function]}
            >
              <a>
                M
              </a>
            </li>
            <li
              key="13"
              onClick={[Function]}
            >
              <a>
                N
              </a>
            </li>
            <li
              key="14"
              onClick={[Function]}
            >
              <a>
                O
              </a>
            </li>
            <li
              key="15"
              onClick={[Function]}
            >
              <a>
                P
              </a>
            </li>
            <li
              key="16"
              onClick={[Function]}
            >
              <a>
                Q
              </a>
            </li>
            <li
              key="17"
              onClick={[Function]}
            >
              <a>
                R
              </a>
            </li>
            <li
              key="18"
              onClick={[Function]}
            >
              <a>
                S
              </a>
            </li>
            <li
              key="19"
              onClick={[Function]}
            >
              <a>
                T
              </a>
            </li>
            <li
              key="20"
              onClick={[Function]}
            >
              <a>
                U
              </a>
            </li>
            <li
              key="21"
              onClick={[Function]}
            >
              <a>
                V
              </a>
            </li>
            <li
              key="22"
              onClick={[Function]}
            >
              <a>
                W
              </a>
            </li>
            <li
              key="23"
              onClick={[Function]}
            >
              <a>
                X
              </a>
            </li>
            <li
              key="24"
              onClick={[Function]}
            >
              <a>
                Y
              </a>
            </li>
            <li
              key="25"
              onClick={[Function]}
            >
              <a>
                Z
              </a>
            </li>
            <li
              onClick={[Function]}
            >
              <FontAwesome
                name="search"
              />
            </li>
          </ul>
        </div>
        <VelocityTransitionGroup
          enter={
            Object {
              "animation": "slideDown",
              "complete": [Function],
              "duration": 250,
            }
          }
          enterHideStyle={
            Object {
              "display": "none",
            }
          }
          enterShowStyle={
            Object {
              "display": "",
            }
          }
          leave={
            Object {
              "animation": "slideUp",
              "duration": 250,
            }
          }
          runOnMount={false}
        />
      </div>
      <div
        className="Artist"
        key="1"
      >
        <Link
          replace={false}
          to="/artist/abc"
        >
          <h2>
            ABC
          </h2>
          <p />
          <span
            className="icon"
          >
            <FontAwesome
              name="music"
            />
             
            2
            <FontAwesome
              className="spacer-left-xs"
              name="comment-o"
            />
             
            50
          </span>
        </Link>
      </div>
    </div>
  </Col>
</Row>
`;

exports[`<Artists /> renders ABC search match 1`] = `
<Row
  bsClass="row"
  componentClass="div"
>
  <Col
    bsClass="col"
    componentClass="div"
    md={10}
  >
    <div
      className="Artists"
    >
      <PageHeader
        bsClass="page-header"
      >
        Artists 
        <small>
          (
          1
           
          Artist
          )
        </small>
      </PageHeader>
      <div>
        <div
          className="filters"
        >
          <ul
            className="pagination pagination-sm"
          >
            <li
              onClick={[Function]}
            >
              <a>
                All
              </a>
            </li>
            <li
              key="0"
              onClick={[Function]}
            >
              <a>
                A
              </a>
            </li>
            <li
              key="1"
              onClick={[Function]}
            >
              <a>
                B
              </a>
            </li>
            <li
              key="2"
              onClick={[Function]}
            >
              <a>
                C
              </a>
            </li>
            <li
              key="3"
              onClick={[Function]}
            >
              <a>
                D
              </a>
            </li>
            <li
              key="4"
              onClick={[Function]}
            >
              <a>
                E
              </a>
            </li>
            <li
              key="5"
              onClick={[Function]}
            >
              <a>
                F
              </a>
            </li>
            <li
              key="6"
              onClick={[Function]}
            >
              <a>
                G
              </a>
            </li>
            <li
              key="7"
              onClick={[Function]}
            >
              <a>
                H
              </a>
            </li>
            <li
              key="8"
              onClick={[Function]}
            >
              <a>
                I
              </a>
            </li>
            <li
              key="9"
              onClick={[Function]}
            >
              <a>
                J
              </a>
            </li>
            <li
              key="10"
              onClick={[Function]}
            >
              <a>
                K
              </a>
            </li>
            <li
              key="11"
              onClick={[Function]}
            >
              <a>
                L
              </a>
            </li>
            <li
              key="12"
              onClick={[Function]}
            >
              <a>
                M
              </a>
            </li>
            <li
              key="13"
              onClick={[Function]}
            >
              <a>
                N
              </a>
            </li>
            <li
              key="14"
              onClick={[Function]}
            >
              <a>
                O
              </a>
            </li>
            <li
              key="15"
              onClick={[Function]}
            >
              <a>
                P
              </a>
            </li>
            <li
              key="16"
              onClick={[Function]}
            >
              <a>
                Q
              </a>
            </li>
            <li
              key="17"
              onClick={[Function]}
            >
              <a>
                R
              </a>
            </li>
            <li
              key="18"
              onClick={[Function]}
            >
              <a>
                S
              </a>
            </li>
            <li
              key="19"
              onClick={[Function]}
            >
              <a>
                T
              </a>
            </li>
            <li
              key="20"
              onClick={[Function]}
            >
              <a>
                U
              </a>
            </li>
            <li
              key="21"
              onClick={[Function]}
            >
              <a>
                V
              </a>
            </li>
            <li
              key="22"
              onClick={[Function]}
            >
              <a>
                W
              </a>
            </li>
            <li
              key="23"
              onClick={[Function]}
            >
              <a>
                X
              </a>
            </li>
            <li
              key="24"
              onClick={[Function]}
            >
              <a>
                Y
              </a>
            </li>
            <li
              key="25"
              onClick={[Function]}
            >
              <a>
                Z
              </a>
            </li>
            <li
              onClick={[Function]}
            >
              <FontAwesome
                name="search"
              />
            </li>
          </ul>
        </div>
        <VelocityTransitionGroup
          enter={
            Object {
              "animation": "slideDown",
              "complete": [Function],
              "duration": 250,
            }
          }
          enterHideStyle={
            Object {
              "display": "none",
            }
          }
          enterShowStyle={
            Object {
              "display": "",
            }
          }
          leave={
            Object {
              "animation": "slideUp",
              "duration": 250,
            }
          }
          runOnMount={false}
        />
      </div>
      <div
        className="Artist"
        key="1"
      >
        <Link
          replace={false}
          to="/artist/abc"
        >
          <h2>
            ABC
          </h2>
          <p />
          <span
            className="icon"
          >
            <FontAwesome
              name="music"
            />
             
            2
            <FontAwesome
              className="spacer-left-xs"
              name="comment-o"
            />
             
            50
          </span>
        </Link>
      </div>
    </div>
  </Col>
</Row>
`;

exports[`<Artists /> renders first page 1`] = `
<Row
  bsClass="row"
  componentClass="div"
>
  <Col
    bsClass="col"
    componentClass="div"
    md={10}
  >
    <div
      className="Artists"
    >
      <PageHeader
        bsClass="page-header"
      >
        Artists 
        <small>
          (
          3
           
          Artists
          )
        </small>
      </PageHeader>
      <div>
        <div
          className="filters"
        >
          <ul
            className="pagination pagination-sm"
          >
            <li
              onClick={[Function]}
            >
              <a>
                All
              </a>
            </li>
            <li
              key="0"
              onClick={[Function]}
            >
              <a>
                A
              </a>
            </li>
            <li
              key="1"
              onClick={[Function]}
            >
              <a>
                B
              </a>
            </li>
            <li
              key="2"
              onClick={[Function]}
            >
              <a>
                C
              </a>
            </li>
            <li
              key="3"
              onClick={[Function]}
            >
              <a>
                D
              </a>
            </li>
            <li
              key="4"
              onClick={[Function]}
            >
              <a>
                E
              </a>
            </li>
            <li
              key="5"
              onClick={[Function]}
            >
              <a>
                F
              </a>
            </li>
            <li
              key="6"
              onClick={[Function]}
            >
              <a>
                G
              </a>
            </li>
            <li
              key="7"
              onClick={[Function]}
            >
              <a>
                H
              </a>
            </li>
            <li
              key="8"
              onClick={[Function]}
            >
              <a>
                I
              </a>
            </li>
            <li
              key="9"
              onClick={[Function]}
            >
              <a>
                J
              </a>
            </li>
            <li
              key="10"
              onClick={[Function]}
            >
              <a>
                K
              </a>
            </li>
            <li
              key="11"
              onClick={[Function]}
            >
              <a>
                L
              </a>
            </li>
            <li
              key="12"
              onClick={[Function]}
            >
              <a>
                M
              </a>
            </li>
            <li
              key="13"
              onClick={[Function]}
            >
              <a>
                N
              </a>
            </li>
            <li
              key="14"
              onClick={[Function]}
            >
              <a>
                O
              </a>
            </li>
            <li
              key="15"
              onClick={[Function]}
            >
              <a>
                P
              </a>
            </li>
            <li
              key="16"
              onClick={[Function]}
            >
              <a>
                Q
              </a>
            </li>
            <li
              key="17"
              onClick={[Function]}
            >
              <a>
                R
              </a>
            </li>
            <li
              key="18"
              onClick={[Function]}
            >
              <a>
                S
              </a>
            </li>
            <li
              key="19"
              onClick={[Function]}
            >
              <a>
                T
              </a>
            </li>
            <li
              key="20"
              onClick={[Function]}
            >
              <a>
                U
              </a>
            </li>
            <li
              key="21"
              onClick={[Function]}
            >
              <a>
                V
              </a>
            </li>
            <li
              key="22"
              onClick={[Function]}
            >
              <a>
                W
              </a>
            </li>
            <li
              key="23"
              onClick={[Function]}
            >
              <a>
                X
              </a>
            </li>
            <li
              key="24"
              onClick={[Function]}
            >
              <a>
                Y
              </a>
            </li>
            <li
              key="25"
              onClick={[Function]}
            >
              <a>
                Z
              </a>
            </li>
            <li
              onClick={[Function]}
            >
              <FontAwesome
                name="search"
              />
            </li>
          </ul>
        </div>
        <VelocityTransitionGroup
          enter={
            Object {
              "animation": "slideDown",
              "complete": [Function],
              "duration": 250,
            }
          }
          enterHideStyle={
            Object {
              "display": "none",
            }
          }
          enterShowStyle={
            Object {
              "display": "",
            }
          }
          leave={
            Object {
              "animation": "slideUp",
              "duration": 250,
            }
          }
          runOnMount={false}
        />
      </div>
      <div
        className="Artist"
        key="1"
      >
        <Link
          replace={false}
          to="/artist/abc"
        >
          <h2>
            ABC
          </h2>
          <p />
          <span
            className="icon"
          >
            <FontAwesome
              name="music"
            />
             
            2
            <FontAwesome
              className="spacer-left-xs"
              name="comment-o"
            />
             
            50
          </span>
        </Link>
      </div>
      <div
        className="Artist"
        key="2"
      >
        <Link
          replace={false}
          to="/artist/def"
        >
          <h2>
            DEF
          </h2>
          <p />
          <span
            className="icon"
          >
            <FontAwesome
              name="music"
            />
             
            4
            <FontAwesome
              className="spacer-left-xs"
              name="comment-o"
            />
             
            2
          </span>
        </Link>
      </div>
      <Paginator
        onPageChange={[Function]}
        pagination={
          Object {
            "current_page": 1,
            "next_page": 2,
            "prev_page": null,
            "total_count": 3,
            "total_pages": 2,
          }
        }
      />
    </div>
  </Col>
</Row>
`;

exports[`<Artists /> renders second page 1`] = `
<Row
  bsClass="row"
  componentClass="div"
>
  <Col
    bsClass="col"
    componentClass="div"
    md={10}
  >
    <div
      className="Artists"
    >
      <PageHeader
        bsClass="page-header"
      >
        Artists 
        <small>
          (
          3
           
          Artists
          )
        </small>
      </PageHeader>
      <div>
        <div
          className="filters"
        >
          <ul
            className="pagination pagination-sm"
          >
            <li
              onClick={[Function]}
            >
              <a>
                All
              </a>
            </li>
            <li
              key="0"
              onClick={[Function]}
            >
              <a>
                A
              </a>
            </li>
            <li
              key="1"
              onClick={[Function]}
            >
              <a>
                B
              </a>
            </li>
            <li
              key="2"
              onClick={[Function]}
            >
              <a>
                C
              </a>
            </li>
            <li
              key="3"
              onClick={[Function]}
            >
              <a>
                D
              </a>
            </li>
            <li
              key="4"
              onClick={[Function]}
            >
              <a>
                E
              </a>
            </li>
            <li
              key="5"
              onClick={[Function]}
            >
              <a>
                F
              </a>
            </li>
            <li
              key="6"
              onClick={[Function]}
            >
              <a>
                G
              </a>
            </li>
            <li
              key="7"
              onClick={[Function]}
            >
              <a>
                H
              </a>
            </li>
            <li
              key="8"
              onClick={[Function]}
            >
              <a>
                I
              </a>
            </li>
            <li
              key="9"
              onClick={[Function]}
            >
              <a>
                J
              </a>
            </li>
            <li
              key="10"
              onClick={[Function]}
            >
              <a>
                K
              </a>
            </li>
            <li
              key="11"
              onClick={[Function]}
            >
              <a>
                L
              </a>
            </li>
            <li
              key="12"
              onClick={[Function]}
            >
              <a>
                M
              </a>
            </li>
            <li
              key="13"
              onClick={[Function]}
            >
              <a>
                N
              </a>
            </li>
            <li
              key="14"
              onClick={[Function]}
            >
              <a>
                O
              </a>
            </li>
            <li
              key="15"
              onClick={[Function]}
            >
              <a>
                P
              </a>
            </li>
            <li
              key="16"
              onClick={[Function]}
            >
              <a>
                Q
              </a>
            </li>
            <li
              key="17"
              onClick={[Function]}
            >
              <a>
                R
              </a>
            </li>
            <li
              key="18"
              onClick={[Function]}
            >
              <a>
                S
              </a>
            </li>
            <li
              key="19"
              onClick={[Function]}
            >
              <a>
                T
              </a>
            </li>
            <li
              key="20"
              onClick={[Function]}
            >
              <a>
                U
              </a>
            </li>
            <li
              key="21"
              onClick={[Function]}
            >
              <a>
                V
              </a>
            </li>
            <li
              key="22"
              onClick={[Function]}
            >
              <a>
                W
              </a>
            </li>
            <li
              key="23"
              onClick={[Function]}
            >
              <a>
                X
              </a>
            </li>
            <li
              key="24"
              onClick={[Function]}
            >
              <a>
                Y
              </a>
            </li>
            <li
              key="25"
              onClick={[Function]}
            >
              <a>
                Z
              </a>
            </li>
            <li
              onClick={[Function]}
            >
              <FontAwesome
                name="search"
              />
            </li>
          </ul>
        </div>
        <VelocityTransitionGroup
          enter={
            Object {
              "animation": "slideDown",
              "complete": [Function],
              "duration": 250,
            }
          }
          enterHideStyle={
            Object {
              "display": "none",
            }
          }
          enterShowStyle={
            Object {
              "display": "",
            }
          }
          leave={
            Object {
              "animation": "slideUp",
              "duration": 250,
            }
          }
          runOnMount={false}
        />
      </div>
      <div
        className="Artist"
        key="3"
      >
        <Link
          replace={false}
          to="/artist/ghi"
        >
          <h2>
            GHI
          </h2>
          <p />
          <span
            className="icon"
          >
            <FontAwesome
              name="music"
            />
             
            7
            <FontAwesome
              className="spacer-left-xs"
              name="comment-o"
            />
             
            23
          </span>
        </Link>
      </div>
      <Paginator
        onPageChange={[Function]}
        pagination={
          Object {
            "current_page": 2,
            "next_page": null,
            "prev_page": 1,
            "total_count": 3,
            "total_pages": 2,
          }
        }
      />
    </div>
  </Col>
</Row>
`;
